channelserver/src/meta.rs

```diff
@@ -0,0 +1,309 @@
+use std::collections::BTreeMap;
+
+use actix::Addr;
+use actix_web::{http, HttpRequest};
+use maxminddb::{self, geoip2::City, MaxMindDBError};
+
+use logging;
+use session::WsChannelSessionState;
+
+// Sender meta data, drawn from the HTTP Headers of the connection counterpart.
+#[derive(Serialize, Debug, Clone)]
+pub struct SenderData {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ua: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub remote: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub city: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub region: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub country: Option<String>,
+}
+
+impl Default for SenderData {
```

Comment:
if SenderData isn't changing later you can kill this manual Default impl for a #[derive(Default)] on the struct (it defaults Option<T> to None)